### DISCUSSION TITLE
Modbus PID Settings

### DISCUSSION THREAD

**Discussion by Zemona** [artisan-roaster-scope/artisan#1537] [Q&A]
I am using a Dongyi 3kg DY roaster. On the config port page, I have the correct inputs for reading BT and ET using bluetooth modbus (yeah that works!)  But I do not know how to figure out the PID slave, register and P, I and D addresses. The roast is not following my background profile.

**Comment by poundy** [comment]
Based on https://artisan-scope.org/machines/dongyi/ there is no PID integration, just logging of BT/ET.
Features: logging of bean temperature (BT) and environmental temperature (ET)

**Comment by Zemona** [comment]
I was sold, temp control with artisan software. So, I may have been duped?

**Reply by poundy** [reply]
"duped" is a strong work. "Confused" probably also fits though. You have complete control over the gas knob yourself, looking at the Artisan graph. That to me says temperature control with Artisan, doesn't it? 

At this point, your issue is with the seller and whether you choose to take it further, is up to you. I don't know of many small cheap machines that offer PID integration TBH.

**Comment by Zemona** [comment]
unless resolved, I will go with duped or defrauded. But I think it can be
resolved. I can clearly see that the roaster is modulating the gas to keep
a very straight linear climb to some set rate. That rate to a set-point is
obviously being done, just not to my set-point, yet.

**Reply by MAKOMO** [reply]
If it can be resolved then only by the makers of the machine. Artisan offers all configuration options, you just need to fill in the correct values that only he can know.

**Reply by Zemona** [reply]
the roaster clearly has a solenoid moderating the gas, might someone might recognize controller type and therfore contribute to finding the SV addresses? 
![IMG_1553](https://github.com/artisan-roaster-scope/artisan/assets/167618005/94849bd7-56fe-47ff-8d25-7267f073752b)

**Reply by poundy** [reply]
yes, the manufacturer of the roaster would. There's an infinite number of PIDs in the world, why would anyone here bother trying to guess when the vendor will KNOW?

**Comment by Terracotta-6** [comment]
http://www.hanyoungnux.com/product/product_view.php?lcode=01&mcode=&pcode=1806100001&scode=0102002#

**Reply by Terracotta-6** [reply]
[manual_eng.pdf](https://github.com/artisan-roaster-scope/artisan/files/15069235/manual_eng.pdf)

**Reply by Terracotta-6** [reply]
[Huong-dan-su-dung-bo-dieu-khien-nhiet-do-Hanyoung-VX-Series.pdf](https://github.com/artisan-roaster-scope/artisan/files/15069706/Huong-dan-su-dung-bo-dieu-khien-nhiet-do-Hanyoung-VX-Series.pdf)

**Reply by Terracotta-6** [reply]
The problem has nothing to do with Artian, the equipment manufacturer consults.If you want to modify the equipment yourself, you need the technical support of engineers, otherwise there will be potential safety hazards.

**Reply by Zemona** [reply]
thank you
